app/uk/gov/hmrc/plasticpackagingtax/registration/models/request/KeepAliveAction.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.models.request
+
+import com.google.inject.ImplementedBy
+import play.api.Logger
+import play.api.mvc.{ActionRefiner, Result}
+import uk.gov.hmrc.auth.core.SessionRecordNotFound
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.plasticpackagingtax.registration.config.AppConfig
+import uk.gov.hmrc.plasticpackagingtax.registration.repositories.RegistrationAmendmentRepository
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class KeepAliveActionImpl @Inject() (
+  appConfig: AppConfig,
+  registrationAmendmentRepository: RegistrationAmendmentRepository
+)(implicit val exec: ExecutionContext)
+    extends KeepAliveAction {
+
+  private val logger = Logger(this.getClass)
+
+  protected def refine[A](
+    request: AuthenticatedRequest[A]
+  ): Future[Either[Result, JourneyRequest[A]]] = {
+    implicit val hc: HeaderCarrier =
+      HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+    request.session.get("sessionId") match {
+      case Some(sessionId) =>
+        registrationAmendmentRepository.get(sessionId).flatMap {
```

Comment:
This will only keep the registration amendment alive. Can we do something more generic with the underlying `UserDataRepository`? Be good to find all a user's documents and keep them all alive.